src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java

```diff
@@ -167,8 +191,40 @@ private JComponent getBody() throws IOException, InvalidUrlException {
         return body;
     }
 
-    private JComponent getFailure() {
-        return new JLabel("Error to get information about this component");
+    public String addChar(String str, String character, int position) {
+        return str.substring(0, ++position) + character + str.substring(position);
+    }
+
+    private JComponent getFailure() throws InvalidUrlException, IOException {
```

Comment:
Return a default JComponent in case of exception. For example, a label which contains the text: "Error". This will avoid to get an empty label and nothing display in the popover.